File: Check If The String Is A Palindrome.py
#https://www.naukri.com/code360/problems/check-if-the-string-is-a-palindrome_1062633

from os import *
from sys import *
from collections import *
from math import *

def checkPalindrome(s):
    # Write your code here
	# Return a boolean
	st=""
	for i in s:
		if (ord(i)>=65 and ord(i)<=65+26) or (ord(i)>=97 and ord(i)<=97+26) or (ord(i)>=48 and ord(i)<=48+10):
			st+=i.upper()
	from_last=len(st)
	#print(st)
	for i in range(len(st)//2):
		from_last-=1
		if st[i]!=st[from_last]:
			return False
	return True

File: Check If The String Is A Palindrome.cpp
#

#include <bits/stdc++.h> 
bool checkPalindrome(string s)
{
    // Write your code here.
    int i=0;
    int j=s.length()-1;
    while(i<j){
        int x=s[i];
        if ((x>=48 && x<=57) || (x>=65 && x<=90) || (x>=97 && x<=122)){
            if (x>=97 && x<=122){
                s[i]=char(x-32);
            }
            int x=s[j];
            if ((x>=48 && x<=57) || (x>=65 && x<=90) || (x>=97 && x<=122)){
                if (x>=97 && x<=122){
                    s[j]=char(x-32);
                }
                // std::cout<<s[i]<<s[j];
                if (s[i]!=s[j]){
                    return 0;
                }
                else{
                    
                    i+=1;
                    j-=1;
                }
            }
            else{
                j-=1;
            }

        
        }
        else{
            i+=1;
        }
    }
    return "Yes";
}
